### ISSUE TITLE
unnecessary_operation false positive with unsafe blocks`

### ISSUE THREAD

**Issue by Manishearth** [Manishearth/rust-clippy#952]
```rust
pub unsafe fn method() {}

pub fn safe_method() {
    unsafe { method() };
}
```

```
lib.rs:4:5: 4:25 warning: statement can be reduced, #[warn(unnecessary_operation)] on by default
lib.rs:4     unsafe { method() };
             ^~~~~~~~~~~~~~~~~~~~
lib.rs:4:5: 4:25 help: replace it with
lib.rs:      method();
lib.rs:4:5: 4:25 help: for further information visit https://github.com/Manishearth/rust-clippy/wiki#unnecessary_operation
```

**Comment by oli-obk** [comment]
whoops, that was me. I'll fix it
